packages/transactions/src/utils.ts

```diff
@@ -182,8 +182,12 @@ export interface ReadOnlyFunctionResponse {
  * Converts the response of a read-only function call into its Clarity Value
  * @param param
  */
-export const parseReadOnlyResponse = ({ result }: ReadOnlyFunctionResponse): ClarityValue => {
-  return hexToCV(result);
+export const parseReadOnlyResponse = ({ okay, result }: ReadOnlyFunctionResponse): ClarityValue => {
+  if (okay) {
+    return hexToCV(result);
+  } else {
+    return responseErrorCV(noneCV());
```

Comment:
I don't think this is right - shouldn't we parse the error and return that?

Reply:
Should be:

```ts
export interface ReadOnlyFunctionSuccessResponse {
  okay: true;
  result: string;
}

export interface ReadOnlyFunctionErrorResponse {
  okay: false;
  cause: string;
}

export type ReadOnlyFunctionResponse = ReadOnlyFunctionSuccessResponse | ReadOnlyFunctionErrorResponse;

/**
 * Converts the response of a read-only function call into its Clarity Value
 * @param param
 */
export const parseReadOnlyResponse = (response: ReadOnlyFunctionResponse): ClarityValue => {
  if (response.okay) {
    return hexToCV(response.result);
  } else {
    throw new Error(response.cause);
  }
};
```

Reply:
Makes sense!